functional-tests/src/main/java/com/redhat/che/selenium/core/workspace/CheStarterWrapper.java

```diff
@@ -11,166 +11,209 @@
-    sb.append(path);
-    sb.append("?");
-    sb.append("masterUrl=").append(this.host).append("&").append("namespace=sthf");
-    Builder requestBuilder = new Request.Builder().url(sb.toString());
-    requestBuilder.addHeader("Authorization", "Bearer " + token);
-    RequestBody body = RequestBody.create(null, new byte[0]);
-    Request request = requestBuilder.patch(body).build();
-    try {
-      Response response = client.newCall(request).execute();
+  public void startWorkspace(String WorkspaceID, String workspaceName, String token)
+      throws Exception {
+    String relativePath = "/workspace/" + workspaceName;
+    String authorization = "Bearer " + token;
+    RequestBody body = RequestBody.create(MediaType.parse("application/json; charset=urf-8"), "{}");
+
+    StringBuilder cheServerURLBuilder = new StringBuilder(this.cheStarterURL);
+    cheServerURLBuilder.append(relativePath);
+    cheServerURLBuilder.append('?');
+    getDefaultQueryParameters()
+        .forEach(
+            (key, value) -> cheServerURLBuilder.append(key).append('=').append(value).append('&'));
+    cheServerURLBuilder.deleteCharAt(cheServerURLBuilder.lastIndexOf("&"));
+
+    Request request =
+        new Request.Builder()
+            .url(cheServerURLBuilder.toString())
+            .addHeader("Authorization", authorization)
+            .patch(body)
+            .build();
+
+    OkHttpClient client = new OkHttpClient.Builder().build();
+    Response response = client.newCall(request).execute();
+    if (response.isSuccessful()) {
+      relativePath = "workspace/" + WorkspaceID + "/runtime";
+      request =
+          new Request.Builder()
+              .url(testApiEndpointUrlProvider.get().toString().concat(relativePath))
+              .addHeader("Authorization", authorization)
+              .post(body)
+              .build();
+      response = client.newCall(request).execute();
       if (response.isSuccessful()) {
-        LOG.info("Prepare workspace request send. Starting workspace.");
-        sb = new StringBuilder("https://" + this.host);
-        sb.append("/api/workspace/");
-        sb.append(WorkspaceID);
-        sb.append("/runtime");
-        requestBuilder = new Request.Builder().url(sb.toString());
-        requestBuilder.addHeader("Authorization", "Bearer " + token);
-        request = requestBuilder.post(body).build();
-        response = client.newCall(request).execute();
-        if (response.isSuccessful()) {
-          LOG.info("Workspace was started. Waiting until workspace is running.");
-        }
+        LOG.info("Workspace was started successfully. Waiting for state RUNNING.");
+        return;
+      } else {
+        LOG.error("Che server start workspace failed:" + response.body().string());
       }
-    } catch (IOException e) {
-      LOG.error("Workspace start failed : " + e.getMessage(), e);
-      throw e;
+    } else {
+      LOG.error("Che starter start workspace failed:" + response.body().string());
```

Comment:
The more accurate message should be handy. Saying something like "Che starter set environment before worskpace start failed:"?